www/scripts/productlist/productlist.js

```diff
@@ -69,29 +71,52 @@ function (declare, topic, domConstruct, Button, BorderContainer,
 
     //--- Admin button ---//
 
-    // TODO: Show admin button only if superuser.
-    /* var menuButton = new Button({
-      class : 'mainMenuButton',
-      label : 'Administration',
+    layout.set('adminLevel', 0);
+
+    var isSuperuser = AUTH_SERVICE.hasPermission(
+      AUTH_OBJECTS.Permission.SUPERUSER, "");
+
+    var isAdminOfAnyProduct = PROD_SERVICE.isAdministratorOfAnyProduct();
+
+    var menuButton = new Button({
```

Comment:
If I'm allow to edit the product list, why don't we show the administration settings (edit, remove icons, create new product button, etc.) by default?

Reply:
It's bit of an extra security measure to prevent an accidental permission override or product deletion. A product deletion is a dangerous action as it irrecoverably **destroys** the access control list of a product. It does **not** destroy analysis results, but after adding the product back, the permission list will be reset to defaults.

Think of it as `sudo` on Linux or the UAC introduced with Windows Vista. Even though you are an admin, it's better to put this extra barrier before dangerous actions.